src/Operations.jl

```diff
@@ -115,34 +114,113 @@ end
 load_package_data(f::Base.Callable, path::String, version::VersionNumber) =
     get(load_package_data(f, path, [version]), version, nothing)
 
-function deps_graph(env::EnvCache, pkgs::Vector{PackageSpec})
-    deps = Dict{UUID,Dict{VersionNumber,Tuple{SHA1,Dict{UUID,VersionSpec}}}}()
+function deps_graph(env::EnvCache, pkgs::Vector{PackageSpec}, reqs)
+    deps = Dict{UUID,Dict{VersionNumber,Dict{UUID,VersionSpec}}}()
     uuids = [pkg.uuid for pkg in pkgs]
     seen = UUID[]
-    while true
-        unseen = setdiff(uuids, seen)
-        isempty(unseen) && break
-        for uuid in unseen
-            push!(seen, uuid)
-            deps[uuid] = valtype(deps)()
-            for path in registered_paths(env, uuid)
-                version_info = load_versions(path)
-                versions = sort!(collect(keys(version_info)))
-                dependencies = load_package_data(UUID, joinpath(path, "dependencies.toml"), versions)
-                compatibility = load_package_data(VersionSpec, joinpath(path, "compatibility.toml"), versions)
-                for (v, h) in version_info
-                    d = get_or_make(Dict{String,UUID}, dependencies, v)
-                    r = get_or_make(Dict{String,VersionSpec}, compatibility, v)
-                    q = Dict(u => get_or_make(VersionSpec, r, p) for (p, u) in d)
-                    # VERSION in get_or_make(VersionSpec, r, "julia") || continue
-                    deps[uuid][v] = (h, q)
-                    for (p, u) in d
-                        u in uuids || push!(uuids, u)
+    for i in 1:2
```

Comment:
This is a bit ugly. I did it like this because I want `collect_fixed` to overwrite the stuff from the dependency resolver above, but `collect_fixed` can itself find new dependencies so we need to run this loop again to fully resolve. Perhaps this can be done in a better way though.